File: docs/sphinx/snippets/python/default_ops/x_op.py
import cudaq

def main():
    qubit = cudaq.qubit()
# [Begin X Op]
    # Apply the unitary transformation
    # X = | 0  1 |
    #     | 1  0 |
    x(qubit)
# [Begin X Op]
if __name__ == "__main__":
    main()

File: docs/sphinx/snippets/cpp/default_ops/x_op.cpp
/*******************************************************************************
 * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
 * All rights reserved.                                                        *
 *                                                                             *
 * This source code and the accompanying materials are made available under    *
 * the terms of the Apache License 2.0 which accompanies this distribution.    *
 ******************************************************************************/

// Compile and run with: `nvq++ x_op.cpp && ./a.out`

#include <cudaq.h>

int main() {
  cudaq::qubit qubit;

  // Apply the unitary transformation
  // X = | 0  1 |
  //     | 1  0 |
  x(qubit);

  return 0;
}

